R/census_helper_v2.R

```diff
@@ -162,49 +145,57 @@ census_helper_new <- function(key, voter.file, states = "all", geo = "tract", ag
                                    # Only those counties within the target state
                                    counties = unique(voter.file$county[voter.file$state == state]))
         } else {
-          census <- census_geo_api(key, state, geo = "block", age, sex, retry)
+          census <- census_geo_api(key, state, geo = "block", age, sex, year, retry)
         }
         
       } else {
         census <- census.data[[toupper(state)]]$block
       }
+    } else {
+      geo.merge <- geo
+      
+      state_must_be_downloaded <- toDownload ||
+        is.null(census.data[[state]]) ||
+        census.data[[state]]$year != year ||
+        # TODO: Why do we always redownload if sex or age == TRUE?
```

Comment:
Hi! I was running the age and sex extensions on predownloaded census data and noticed that all the census data is being pulled again. Did anyone look into why this happens?

Reply:
It can happen if you are missing some combination of data. For example, if you change age/sex variables it may redownload. If you have a new row in the voter data with a state or county or something else it may trigger a redownload. 

I tried a couple of different scenarios and it appears to be working for me, but if you can recreate your problem, please submit an issue with a reprex using 3.0.0 code base.